### PR TITLE
 add ZKPathConstantTest (#484)

### PR DESCRIPTION
**Pull request by songyuequan**
Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+/**
+ * Test cases for ZkPathConstants.
+ *
+ * @author songyuequan
+ */
+public class ZkPathConstantsTest {
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:31]: add final

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Test
+    public void testBuildAppAuthPath() {
+        String appKey = RandomStringUtils.randomAlphanumeric(10);
+        assertNotNull(ZkPathConstants.buildAppAuthPath(appKey));
+        assertEquals("/soul/auth/" + appKey, ZkPathConstants.buildAppAuthPath(appKey));
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:37]: 这些字符串抽成公共的变量

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    @Test
+    public void testBuildMetaDataPath() {
+        String metadata = RandomStringUtils.randomAlphanumeric(10);
+        assertNotNull(ZkPathConstants.buildMetaDataPath(metadata));
+        assertEquals("/soul/metaData/" + metadata, ZkPathConstants.buildMetaDataPath(metadata));
+    }
+
+    @Test
+    public void testBuildPluginParentPath() {
+        assertEquals("/soul/plugin", ZkPathConstants.buildPluginParentPath());
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:49]: /soul/plugin   /soul/selector 这些抽查公共的变量

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Test
+    public void testBuildPluginPath() {
+        String pluginName = RandomStringUtils.randomAlphanumeric(10);
+        assertEquals("/soul/plugin/" + pluginName, ZkPathConstants.buildPluginPath(pluginName));
+        assertEquals(ZkPathConstants.buildPluginParentPath() + "/" + pluginName, ZkPathConstants.buildPluginPath(pluginName));
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:56]: 字符串拼接使用String.join 不要使用 +

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        String pluginName = RandomStringUtils.randomAlphanumeric(10);
+        assertEquals("/soul/rule/" + pluginName, ZkPathConstants.buildRuleParentPath(pluginName));
+    }
+
+    @Test
+    public void testBuildRulePath() {
+        String pluginName = RandomStringUtils.randomAlphanumeric(10);
+        String selectorId = RandomStringUtils.randomAlphanumeric(10);
+        String ruleId = RandomStringUtils.randomAlphanumeric(10);
+        assertEquals("/soul/rule/" + pluginName + "/" + selectorId + "-" + ruleId, ZkPathConstants.buildRulePath(pluginName, selectorId, ruleId));
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:83]: 字符串拼接使用string.join

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    public static final String SELECTOR_PARENT = "/soul/selector";
+
+    public static final String RULE_PARENT = "/soul/rule";
+
+    public static final String SELECTOR_JOIN_RULE = "-";
+
+    @Test
+    public void testBuildAppAuthPath() {
+        String appKey = RandomStringUtils.randomAlphanumeric(10);
+        assertNotNull(ZkPathConstants.buildAppAuthPath(appKey));
```
Comment (dengliming) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:48]: 这个可以定义到外面只调用一次吗，其他测试类似

Comment (songyuequan) [reply soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:48]: OK

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+ */
+
+package org.dromara.soul.common.constant;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:25]: remove empty line

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        assertNotNull(appAuthPath);
+        assertEquals(String.join("/", APP_AUTH_PARENT, appKey), appAuthPath);
+    }
+
+    @Test
+    public void testBuildMetaDataPath() {
+        String metadata = RandomStringUtils.randomAlphanumeric(10);
+        String metaDataPath = ZkPathConstants.buildMetaDataPath(metadata);
+        assertNotNull(metaDataPath);
+        assertEquals(String.join("/", META_DATA_PARENT, metadata), metaDataPath);
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:58]: "/" maybe private static final constant

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Test
+    public void testBuildRulePath() {
+        String pluginName = RandomStringUtils.randomAlphanumeric(10);
+        String selectorId = RandomStringUtils.randomAlphanumeric(10);
+        String ruleId = RandomStringUtils.randomAlphanumeric(10);
+        String rulePath = ZkPathConstants.buildRulePath(pluginName, selectorId, ruleId);
+        assertNotNull(rulePath);
+        assertEquals(String.join("/", RULE_PARENT, pluginName, selectorId + SELECTOR_JOIN_RULE + ruleId), rulePath);
```
Comment (yu199195) [inline soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:109]: string " +" repalce String.join()

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java (hunk 7 above)
Comment (songyuequan) [reply soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java:25]: OK

**Review by dengliming — APPROVED** [review]

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java (hunk 9 above)
Comment (songyuequan) [reply soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java]: i have changed. it.

File: soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java (hunk 8 above)
Comment (songyuequan) [reply soul-common/src/test/java/org/dromara/soul/common/constant/ZkPathConstantsTest.java]: i have changed. it.

**Pull request merged**